packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -943,45 +943,32 @@ class RenderClipRect extends _RenderCustomClip<Rect> {
 
 /// Clips its child using a rounded rectangle.
 ///
-/// Creates a rounded rectangle from its layout dimensions and the given x and
-/// y radius values and prevents its child from painting outside that rounded
+/// Creates a rounded rectangle from its layout dimensions and the given border
+/// radius and prevents its child from painting outside that rounded
 /// rectangle.
 class RenderClipRRect extends RenderProxyBox {
   /// Creates a rounded-rectangular clip.
+  ///
+  /// The [borderRadius] defaults to [BorderRadius.zero], i.e. a rectangle with
+  /// right-angled corners.
   RenderClipRRect({
     RenderBox child,
-    double xRadius,
-    double yRadius
-  }) : _xRadius = xRadius, _yRadius = yRadius, super(child) {
-    assert(_xRadius != null);
-    assert(_yRadius != null);
+    BorderRadius borderRadius: BorderRadius.zero
+  }) : _borderRadius = borderRadius, super(child) {
+    assert(_borderRadius != null);
   }
 
-  /// The radius of the rounded corners in the horizontal direction in logical pixels.
+  /// The border radius of the rounded corners..
   ///
   /// Values are clamped to be between zero and half the width of the render
```

Comment:
For the sake of hit testing and rendering the sum of the horizontal border radius components are clamped to between zero and half renderer's width. Similarly the vertical border radius components are clamped to the renderer's height.